controller/src/functions/localization.ts

```diff
@@ -0,0 +1,154 @@
+import * as dbSchema from "../db/schema/schema";
+import {eq, sql, and, ilike, isNotNull} from "drizzle-orm";
+import {z} from "zod";
+const db = startDb();
+const table = insertSchemas.localization.table;
+
+// you can run azrue crons  manually by invoking
+// http://localhost:7071/admin/functions/manageLocalizationTable.  (or the function name below). You must use a post request, header of Content-Type application/json, and the body as follows { "input": "anything"}. The body doesn't actually matter
+export async function populateLocalization(
+  myTimer: Timer,
+  context: InvocationContext
+): Promise<void> {
+  context.log("Timer function processed request.");
+  const bookNamesResult = await populateScripturalBookNames();
+  context.log(
+    Array.isArray(bookNamesResult)
+      ? `inserted ${bookNamesResult.length} rows of book names`
+      : bookNamesResult.message
+  );
+  const resourceTypesResult = await populationResourceTypes();
+  context.log(
+    Array.isArray(resourceTypesResult)
+      ? `inserted ${resourceTypesResult.length} rows of resource types`
+      : resourceTypesResult.message
+  );
+}
+
+async function populationResourceTypes() {
+  const payload = localizations.reduce(
+    (acc: insertLocalizationType[], curr) => {
+      const rows = Object.entries(curr.dict).map(([key, value]) => {
+        return {ietfCode: curr.ietf, key, value};
+      });
+      acc.push(...rows);
+      return acc;
+    },
+    []
+  );
+  // no need for a transaction since its just the one sql action
+  const res = await polymorphicInsert({
+    tableKey: "localization",
+    content: payload,
+    onConflictDoUpdateArgs: {
+      target: [table.ietfCode, table.key],
+      // loops through every column in given table setting the column to be the value of the excluded (e.g. conflicting) row except for those given in the second argument. For localization though, it just updates the value.
+      set: onConflictSetAllFieldsToSqlExcluded(table, ["ietfCode", "key"]),
+    },
+  });
+  return res;
+}
+
+async function populateScripturalBookNames() {
+  const {scripturalRenderingMetadata, rendering, content, language, gitRepo} =
+    dbSchema;
+  //@ this is the raw sql version of what the orm below is doing
+  // const query = sql.raw(`SELECT book_name, book_slug, ietf_code, id
+  // FROM (SELECT book_name, book_slug, l.ietf_code, c.id,
+  // ROW_NUMBER() OVER (PARTITION BY l.ietf_code, book_slug ORDER BY book_slug) AS rn
+  //     FROM scriptural_rendering_metadata AS srm
+  //     JOIN rendering AS r ON r.id = srm.rendering_id
+  //     JOIN content AS c ON r.content_id = c.id
+  //     JOIN language AS l ON l.ietf_code = c.language_id
+  //     JOIN git_repo AS gr ON c.git_id = gr.id
+  //     WHERE gr.username ILIKE '%wa-catalog%'
+  //     AND c.domain = 'scripture'
+  //     AND book_slug IS NOT NULL
+  // ) AS subquery
+  // WHERE rn = 1
+  // ORDER BY ietf_code, book_slug;`);
```

Comment:
@PurpleGuitar @danparisd - I've done it with the ORM instead of raw sql, but I imaigne the commented out raw sql might be more readable.  In short, this query has those hardcoded dependencies on the git repo username being wa-catalog there. We only need one localized version for each slug (e.g. Gen, Exo), hence the select from rn =1  for each partition. 

The result looks like this: 
![image](https://github.com/WycliffeAssociates/languageapi/assets/67284402/1c101518-1a82-47c6-955a-b30077279284)